Userland/Shell/Shell.cpp

```diff
@@ -940,11 +939,21 @@ void Shell::execute_process(Vector<char const*>&& argv)
         }
         if (saved_errno == ENOENT) {
             do {
-                auto file_result = Core::DeprecatedFile::open(argv[0], Core::OpenMode::ReadOnly);
+                auto path_as_string_or_error = String::from_deprecated_string(argv[0]);
```

Comment:
I intentionally didn't do that in the initial PR, because technically paths are not necessarily valid UTF-8, so this causes Shell to refuse to execute programs whose names are invalid UTF-8. It kinda shouldn't, but I don't have a strong opinion on this.